### PR TITLE
Mae filter

### PR DESCRIPTION
**Pull request by nickhsmith**

### CODE REVIEW

File: drop/modules/mae-pipeline/MAE/filterSNVs.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ if [ $vcf_id != 'QC' ]; then sample_flag="-s ${vcf_id}"; fi
 $bcftools view  $vcf_file | \
     grep -vP '^##INFO=' | \
     awk -F'\t' 'BEGIN {OFS = FS} { if($1 ~ /^[^#]/){ $8 = "." }; print $0 }' | \
+    $bcftools norm -m-both | \
```
Comment (nickhsmith) [inline drop/modules/mae-pipeline/MAE/filterSNVs.sh:30]: split multiple variants on 1 line into a single variant call per line
A > C,G 
into
A > C
A > G

File: drop/config/submodules/MonoallelicExpression.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def __init__(
         super().__init__(config, sampleAnnotation, processedDataDir, processedResultsDir, workDir)
         self.CONFIG_KEYS = [
             "groups", "genome", "qcVcf", "qcGroups", "gatkIgnoreHeaderCheck", "padjCutoff",
-            "allelicRatioCutoff", "maxAF", "gnomAD"
+            "allelicRatioCutoff", "maxAF", "gnomAD","maxVarFreqCohort"
```
Comment (nickhsmith) [inline drop/config/submodules/MonoallelicExpression.py:20]: include the maxVarFreqCohort into known config keys

File: drop/config/submodules/MonoallelicExpression.py [hunk 3]
```diff
@@ -173,7 +173,10 @@ def setGenomeDict(self, genomeFiles):
     # look up for a sampleID genomeFiles{ncbi -> path} and sampleGenomes {sampleID -> ncbi}
     def getGenomePath(self, sampleID):
         try:
-            return self.genomeFiles[self.sampleGenomes[sampleID]]
+            if len(self.genomeFiles) == 1:
+                return list(self.genomeFiles.values())[0]
+            else:
+                return self.genomeFiles[self.sampleGenomes[sampleID]]
```
Comment (nickhsmith) [inline drop/config/submodules/MonoallelicExpression.py:179]: corner case if the genome is defined as a dictionary but only a single value

File: drop/modules/mae-pipeline/MAE/Datasets.R [hunk 4]
```diff
@@ -24,7 +24,7 @@ devNull <- sapply(datasets, function(name){
     cat(paste0(
       "<h1>Dataset: ", name, "</h1>",
       "<p>",
-      "</br>", "<a href='MAE/", name, "--", version, "_results.html'   >MAE results</a>",
+      "</br>", "<a href='MonoallelicExpression/", name, "--", version, "_results.html'   >MAE results</a>",
```
Comment (nickhsmith) [inline drop/modules/mae-pipeline/MAE/Datasets.R:27]: the name has been renamed

File: drop/modules/mae-pipeline/MAE/Results.R [hunk 5]
```diff
@@ -77,12 +78,20 @@ res[, c('aux') := NULL]
 # Bring gene_name column front
 res <- cbind(res[, .(gene_name)], res[, -"gene_name"])
 
+# Calculate variant frequency within cohort 
+maxCohortFreq <-snakemake@params$maxCohortFreq
+res[, N_var := .N, by = .(gene_name, contig, position)]
+res[, cohort_freq := N_var / uniqueN(ID)]
+
+res[,rare := (rare | is.na(rare)) & cohort_freq <= maxCohortFreq] 
+
```
Comment (nickhsmith) [inline drop/modules/mae-pipeline/MAE/Results.R:89]: calculate the cohort frequency and set rare to take gnomAD and cohortFreq into consideration

File: drop/modules/mae-pipeline/MAE/Results.R [hunk 6]
```diff
@@ -125,6 +134,12 @@ ggplot(melt_dt, aes(variable, value)) + geom_boxplot() +
   labs(y = 'Heterozygous SNVs per patient', x = '') +
     annotation_logticks(sides = "l")
 
+#'
+#' ## Variant Frequency within Cohort Histogram
+ggplot(res,aes(x = cohort_freq)) + geom_histogram( binwidth = 0.02)  +
+  geom_vline(xintercept = maxCohortFreq +.01 ,col = "red") +
+  xlab("Variant frequency in cohort") + ylab("Count")
+
```
Comment (nickhsmith) [inline drop/modules/mae-pipeline/MAE/Results.R]: plot a cohortFreq histogram with a red line for the cutoff value

File: drop/modules/mae-pipeline/MAE/Results.R [hunk 7]
```diff
@@ -77,12 +78,20 @@ res[, c('aux') := NULL]
...
+res[, cohort_freq := N_var / uniqueN(ID)]
+
+res[,rare := (rare | is.na(rare)) & cohort_freq <= maxCohortFreq] 
+
 # Add significance columns
 allelicRatioCutoff <- snakemake@params$allelicRatioCutoff
 res[, MAE := padj <= snakemake@params$padjCutoff &
-      (altRatio >= allelicRatioCutoff | altRatio <= (1-allelicRatioCutoff))] 
+      (altRatio >= allelicRatioCutoff | altRatio <= (1-allelicRatioCutoff)) &
+      rare
```
Comment (nickhsmith) [inline drop/modules/mae-pipeline/MAE/Results.R]: include the rare column (influenced by gnomAD and cohort_freq) to make it into the final results

File: drop/modules/mae-pipeline/MAE/Results.R (hunk 6 above)
Comment (vyepez88) [reply drop/modules/mae-pipeline/MAE/Results.R]: cohort_freq should be unique

File: drop/modules/mae-pipeline/MAE/Results.R (hunk 7 above)
Comment (vyepez88) [reply drop/modules/mae-pipeline/MAE/Results.R]: remove rare

**Review by vyepez88 — APPROVED** [review]

**Pull request merged**